RockLib.Configuration.Conditional/ConditionalConfigurationSection.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace RockLib.Configuration.Conditional;
+
+/// <summary>
+/// A ConfigurationSection that can use an alternate ConfigurationSection to
+/// override some values.
+/// </summary>
+public class ConditionalConfigurationSection : IConfigurationSection
```

Comment:
If there isn't an intention to inherit from a class, we should mark it as `sealed`.

Reply:
Good suggestion. This implementation has some intricate logic, and there'd be a high risk of breaking something in a derived class.